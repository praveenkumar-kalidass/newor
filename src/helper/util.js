import CONSTANT from 'constant';

export const formatCurrency = (value) => {
  const formatter = new Intl.NumberFormat('hi-IN', { style: 'currency', currency: 'INR' });
  return formatter.format(value);
};

export const unformatCurrency = (value) => {
  const withoutSymbol = value.replace(CONSTANT.APP_LITERAL.RUPEE_SYMBOL, '');
  const currency = withoutSymbol.replace(/,/g, '');
  return parseFloat(currency);
};

export const formatUrlParams = (data) => Object.keys(data).map((key) => (
  `${encodeURIComponent(key)}=${encodeURIComponent(data[key])}`
)).join('&');
