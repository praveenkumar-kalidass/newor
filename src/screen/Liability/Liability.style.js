import styled from 'styled-components/native';
import { Box, HStack, Text } from 'native-base';

import COLOR from 'constant/color';

export const Container = styled(Box).attrs(() => ({
  p: 5,
}))``;

export const LiabilityListCard = styled(HStack).attrs(({ theme }) => ({
  bg: theme.color.MODAL,
  rounded: 10,
  p: 4,
  space: 5,
  alignItems: 'center',
}))``;

export const LiabilityValue = styled(Text).attrs(() => ({
  color: COLOR.DARK_BACKGROUND_100,
  fontSize: 32,
  bold: true,
}))``;
