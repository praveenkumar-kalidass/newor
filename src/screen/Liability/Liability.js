import React, { useCallback, useEffect, useState } from 'react';
import { TouchableOpacity } from 'react-native';
import { useIsFocused, useNavigation } from '@react-navigation/native';
import FontAwesome5 from 'react-native-vector-icons/FontAwesome5';
import {
  ScrollView, Divider, HStack, VStack, Avatar, Text, Box, Skeleton, Image,
} from 'native-base';
import FontAwesome from 'react-native-vector-icons/FontAwesome';

import COLOR from 'constant/color';
import CONSTANT from 'constant';
import ROUTE from 'constant/route';
import NEWOR_FAILURE from 'asset/image/newor-failure.png';
import withBackground from 'helper/withBackground';
import useTranslation from 'translation/useTranslation';
import Translation from 'translation/Translation';
import useTheme from 'theme/useTheme';
import { Background3 } from 'component/Background';
import AppButton from 'component/AppButton';
import useUser from 'provider/User/useUser';
import useLiability from 'api/useLiability';
import { formatCurrency } from 'helper/util';
import {
  Container, LiabilityCard, LiabilityValue, LiabilityListCard,
} from './Liability.style';

const Liability = () => {
  const [isLoading, setIsLoading] = useState(false);
  const [errorCode, setErrorCode] = useState('');
  const [liabilities, setLiabilities] = useState([]);
  const navigation = useNavigation();
  const isFocused = useIsFocused();
  const { liability, loadLiability } = useUser();
  const { translate } = useTranslation();
  const theme = useTheme();
  const { getLiability } = useLiability();

  const doGetLiability = useCallback(async () => {
    try {
      setErrorCode('');
      setIsLoading(true);
      const { data } = await getLiability(liability.id);
      setLiabilities(data.list);
      loadLiability({
        id: data.id,
        value: data.value,
      });
      setIsLoading(false);
    } catch (error) {
      setIsLoading(false);
      setErrorCode(error?.response?.data?.code || 'NEWOR_INTERNAL_SERVER_ERROR');
    }
  }, [liability]);

  useEffect(() => {
    if (isFocused) {
      doGetLiability(liability.id);
      return;
    }
    setLiabilities([]);
    setErrorCode('');
  }, [isFocused]);

  return (
    <Container>
      <LiabilityCard>
        <VStack justifyContent="space-around" flex={1}>
          <VStack my={3} px={3}>
            <HStack justifyContent="flex-start">
              <Text color={COLOR.LIGHT_BACKGROUND_0}>
                {translate('TOTAL_LIABILITY_VALUE')}
                ,
              </Text>
            </HStack>
            <HStack justifyContent="center">
              <LiabilityValue>{liability.label}</LiabilityValue>
            </HStack>
          </VStack>
          <Divider />
          <HStack flex={1} px={5} alignItems="center" justifyContent="space-around">
            <VStack space={1} alignItems="center" opacity={0}>
              <Text mt={1} fontSize={36} color={COLOR.LIGHT_SUCCESS_100}>2.50%</Text>
              <Text color={COLOR.LIGHT_BACKGROUND_0}>Secured Growth Rate</Text>
            </VStack>
            <TouchableOpacity testID="liability-add" onPress={() => navigation.navigate(ROUTE.LIABILITY_TYPE)}>
              <VStack space={1} alignItems="center">
                <Box borderWidth={1} p={1} rounded="full" borderColor={COLOR.GREEN_BLUE}>
                  <Avatar bg={COLOR.GREEN_BLUE}>
                    <FontAwesome5 color={COLOR.LIGHT_BACKGROUND_100} size={20} name="file-invoice-dollar" />
                  </Avatar>
                </Box>
                <Translation tkey="ADD_LIABILITY" as={Text} color={COLOR.LIGHT_BACKGROUND_0} />
              </VStack>
            </TouchableOpacity>
          </HStack>
        </VStack>
      </LiabilityCard>
      <ScrollView>
        <VStack mt={isLoading ? 16 : 4} mb={200} space={4}>
          <If condition={isLoading}>
            <For each="loader" of={[1, 2, 3]}>
              <LiabilityListCard key={`liability-loader-${loader}`}>
                <Skeleton size={12} rounded="full" startColor={theme.color.BACKGROUND_50} />
                <Skeleton.Text flex={1.5} startColor={theme.color.BACKGROUND_50} />
                <Skeleton size={3} flex={1} rounded="full" startColor={theme.color.BACKGROUND_50} />
              </LiabilityListCard>
            </For>
          </If>
          <If condition={!isLoading}>
            <For each="liability" index="index" of={liabilities}>
              <LiabilityListCard key={`liability-${index}`}>
                <Avatar
                  bg={COLOR.RANDOM[index % COLOR.RANDOM.length]}
                >
                  {CONSTANT.APP_LITERAL.RUPEE_SYMBOL}
                </Avatar>
                <VStack flex={1}>
                  <Translation tkey={liability.type} as={Text} bold />
                  <Translation tkey={liability.lenderName} as={Text} />
                </VStack>
                <HStack flex={1} justifyContent="flex-end">
                  <Text bold>{formatCurrency(liability.value)}</Text>
                </HStack>
              </LiabilityListCard>
            </For>
          </If>
          <If condition={errorCode}>
            <LiabilityListCard justifyContent="center">
              <VStack alignItems="center">
                <Image
                  testID="liability-failed-image"
                  alt={translate('ERROR_CODE.NEWOR_INTERNAL_SERVER_ERROR')}
                  source={NEWOR_FAILURE}
                  size="2xl"
                />
                <Translation tkey={`ERROR_CODE.${errorCode}`} as={Text} />
                <Translation
                  mt={4}
                  variant="secondary"
                  tkey="RELOAD"
                  as={AppButton}
                  endIcon={<FontAwesome size={20} color={COLOR.LIGHT_BACKGROUND_100} name="undo" />}
                  onPress={doGetLiability}
                  testID="liability-reload"
                />
              </VStack>
            </LiabilityListCard>
          </If>
          <If condition={!isLoading && !errorCode && !liabilities.length}>
            <LiabilityListCard justifyContent="center">
              <Translation tkey="NO_LIABILITY_FOUND" as={Text} />
            </LiabilityListCard>
          </If>
        </VStack>
      </ScrollView>
    </Container>
  );
};

export default withBackground({ Component: Liability, Background: Background3 });
