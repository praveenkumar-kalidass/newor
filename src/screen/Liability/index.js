import Liability from './Liability';

export default Liability;
