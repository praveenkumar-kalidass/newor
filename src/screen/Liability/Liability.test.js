import React from 'react';

import { render } from 'test/util';
import Liability from './Liability';

jest.mock('provider/User/useUser', () => () => ({
  user: {},
  liability: {
    id: 'test_liability_id',
    value: 12345.12,
    label: 'â‚¹12,345.12',
  },
}));

describe('Liability', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match snapshot', () => {
    const container = render(<Liability />);

    expect(container).toMatchSnapshot();
  });
});
