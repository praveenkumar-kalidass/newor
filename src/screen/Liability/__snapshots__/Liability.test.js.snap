// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Liability should match snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Array [
          Object {
            "borderWidth": 0,
            "left": 0,
            "position": "absolute",
            "top": 0,
          },
          Object {
            "height": 1334,
            "width": 750,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "position": "absolute",
            },
          ]
        }
      >
        <RNSVGSvgView
          bbHeight={1334}
          bbWidth={750}
          focusable={false}
          height={1334}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "flex": 0,
                "height": 1334,
                "width": 750,
              },
            ]
          }
          width={750}
        >
          <RNSVGGroup>
            <RNSVGPath
              d="M 0,0
      L 0,266.8
      C 50 316.8, 700 316.8, 750 266.8
      L 750,0
      Z"
              fill={4278199178}
              propList={
                Array [
                  "fill",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
      <View
        dataSet={Object {}}
        style={
          Array [
            Object {
              "paddingBottom": 20,
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 20,
            },
            Array [
              Object {},
            ],
          ]
        }
      >
        <View
          dataSet={Object {}}
          elevation={1}
          shadowColor="#000000"
          shadowOffset={
            Object {
              "height": 1,
              "width": 0,
            }
          }
          shadowOpacity={0.8}
          shadowRadius={1}
          style={
            Array [
              Object {
                "backgroundColor": "#ffffff",
                "borderRadius": 10,
                "height": 200,
              },
              Array [
                Object {},
              ],
            ]
          }
        >
          <View
            dataSet={Object {}}
            style={
              Object {
                "flex": 1,
                "justifyContent": "space-around",
              }
            }
          >
            <View
              dataSet={Object {}}
              style={
                Object {
                  "marginBottom": 12,
                  "marginTop": 12,
                  "paddingLeft": 12,
                  "paddingRight": 12,
                }
              }
            >
              <View
                dataSet={Object {}}
                style={
                  Object {
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                  }
                }
              >
                <Text
                  dataSet={Object {}}
                  style={
                    Object {
                      "backgroundColor": undefined,
                      "color": "#000000",
                      "fontFamily": undefined,
                      "fontSize": 14,
                      "fontStyle": "normal",
                      "fontWeight": "400",
                      "letterSpacing": 0,
                      "lineHeight": 21,
                      "textDecorationLine": undefined,
                    }
                  }
                >
                  TOTAL_LIABILITY_VALUE
                  ,
                </Text>
              </View>
              <View
                dataSet={Object {}}
                style={Object {}}
              />
              <View
                dataSet={Object {}}
                style={
                  Object {
                    "flexDirection": "row",
                    "justifyContent": "center",
                  }
                }
              >
                <Text
                  dataSet={Object {}}
                  style={
                    Array [
                      Object {
                        "backgroundColor": undefined,
                        "color": "#000000",
                        "fontFamily": undefined,
                        "fontSize": 24,
                        "fontStyle": "normal",
                        "fontWeight": "700",
                        "letterSpacing": 0,
                        "lineHeight": 36,
                        "textDecorationLine": undefined,
                      },
                      Array [
                        Object {},
                      ],
                    ]
                  }
                >
                  â‚¹12,345.12
                </Text>
              </View>
            </View>
            <View
              dataSet={Object {}}
              style={Object {}}
            />
            <View
              aria-orientation="horizontal"
              dataSet={Object {}}
              style={
                Object {
                  "backgroundColor": "#d4d4d4",
                  "height": 1,
                  "width": "100%",
                }
              }
              thickness="1"
            />
            <View
              dataSet={Object {}}
              style={Object {}}
            />
            <View
              dataSet={Object {}}
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-around",
                  "paddingLeft": 20,
                  "paddingRight": 20,
                }
              }
            >
              <View
                dataSet={Object {}}
                style={
                  Object {
                    "alignItems": "center",
                    "opacity": 0,
                  }
                }
              >
                <Text
                  dataSet={Object {}}
                  style={
                    Object {
                      "backgroundColor": undefined,
                      "color": "#2e7d32",
                      "fontFamily": undefined,
                      "fontSize": 36,
                      "fontStyle": "normal",
                      "fontWeight": "400",
                      "letterSpacing": 0,
                      "lineHeight": 54,
                      "marginTop": 4,
                      "textDecorationLine": undefined,
                    }
                  }
                >
                  2.50%
                </Text>
                <View
                  dataSet={Object {}}
                  style={
                    Object {
                      "height": 4,
                    }
                  }
                />
                <Text
                  dataSet={Object {}}
                  style={
                    Object {
                      "backgroundColor": undefined,
                      "color": "#000000",
                      "fontFamily": undefined,
                      "fontSize": 14,
                      "fontStyle": "normal",
                      "fontWeight": "400",
                      "letterSpacing": 0,
                      "lineHeight": 21,
                      "textDecorationLine": undefined,
                    }
                  }
                >
                  Secured Growth Rate
                </Text>
              </View>
              <View
                dataSet={Object {}}
                style={Object {}}
              />
              <View
                accessible={true}
                collapsable={false}
                focusable={false}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
                testID="liability-add"
              >
                <View
                  dataSet={Object {}}
                  style={
                    Object {
                      "alignItems": "center",
                    }
                  }
                >
                  <View
                    borderColoROr="#8E44AD"
                    dataSet={Object {}}
                    style={
                      Object {
                        "borderRadius": 9999,
                        "borderWidth": 1,
                        "paddingBottom": 4,
                        "paddingLeft": 4,
                        "paddingRight": 4,
                        "paddingTop": 4,
                      }
                    }
                  >
                    <View
                      dataSet={Object {}}
                      style={
                        Object {
                          "alignItems": "center",
                          "backgroundColor": "#8E44AD",
                          "borderColor": "#27272a",
                          "borderRadius": 9999,
                          "height": 48,
                          "justifyContent": "center",
                          "position": "relative",
                          "width": 48,
                        }
                      }
                    >
                      <Text
                        allowFontScaling={false}
                        selectable={false}
                        style={
                          Array [
                            Object {
                              "color": "#ffffff",
                              "fontSize": 20,
                            },
                            undefined,
                            Object {
                              "fontFamily": "FontAwesome5Free-Solid",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                            Object {
                              "fontWeight": "900",
                            },
                          ]
                        }
                      >
                        ï•±
                      </Text>
                    </View>
                  </View>
                  <View
                    dataSet={Object {}}
                    style={
                      Object {
                        "height": 4,
                      }
                    }
                  />
                  <Text
                    dataSet={Object {}}
                    style={
                      Object {
                        "backgroundColor": undefined,
                        "color": "#000000",
                        "fontFamily": undefined,
                        "fontSize": 14,
                        "fontStyle": "normal",
                        "fontWeight": "400",
                        "letterSpacing": 0,
                        "lineHeight": 21,
                        "textDecorationLine": undefined,
                      }
                    }
                  >
                    ADD_LIABILITY
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
        <RCTScrollView
          contentContainerStyle={
            Array [
              Object {},
              Object {
                "dataSet": Object {},
              },
            ]
          }
          dataSet={Object {}}
          style={Object {}}
        >
          <View>
            <View
              dataSet={Object {}}
              style={
                Object {
                  "marginBottom": 200,
                  "marginTop": 16,
                }
              }
            >
              <View
                dataSet={Object {}}
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#eeeeee",
                      "borderRadius": 10,
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "paddingBottom": 16,
                      "paddingLeft": 16,
                      "paddingRight": 16,
                      "paddingTop": 16,
                    },
                    Array [
                      Object {},
                    ],
                  ]
                }
              >
                <Text
                  dataSet={Object {}}
                  style={
                    Object {
                      "backgroundColor": undefined,
                      "color": "#171717",
                      "fontFamily": undefined,
                      "fontSize": 14,
                      "fontStyle": "normal",
                      "fontWeight": "400",
                      "letterSpacing": 0,
                      "lineHeight": 21,
                      "textDecorationLine": undefined,
                    }
                  }
                >
                  NO_LIABILITY_FOUND
                </Text>
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
      <View
        pointerEvents="none"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            undefined,
          ]
        }
      />
    </View>
    <View
      elevation={0}
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      style={
        Array [
          Object {
            "borderWidth": 0,
            "left": -600,
            "position": "absolute",
            "top": 0,
          },
          Object {
            "height": 1334,
            "width": 600,
          },
        ]
      }
    >
      <View
        dataSet={Object {}}
        style={
          Array [
            Object {
              "backgroundColor": "#eeeeee",
              "flex": 1,
              "justifyContent": "space-between",
              "paddingBottom": 80,
              "paddingTop": 80,
            },
            Array [
              Object {},
            ],
          ]
        }
      >
        <View
          dataSet={Object {}}
          style={
            Array [
              Object {
                "borderBottomWidth": 1,
                "borderColor": "#7f7f7f",
                "paddingBottom": 8,
                "paddingLeft": 16,
                "paddingRight": 20,
                "paddingTop": 8,
              },
              Array [
                Object {},
              ],
            ]
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              dataSet={Object {}}
              style={
                Object {
                  "alignItems": "center",
                }
              }
            >
              <View
                dataSet={Object {}}
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#a1a1aa",
                    "borderColor": "#27272a",
                    "borderRadius": 9999,
                    "height": 96,
                    "justifyContent": "center",
                    "position": "relative",
                    "width": 96,
                  }
                }
              >
                <Text
                  allowFontScaling={false}
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": undefined,
                        "fontSize": 40,
                      },
                      undefined,
                      Object {
                        "fontFamily": "FontAwesome",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  ïˆ›
                </Text>
              </View>
              <View
                dataSet={Object {}}
                style={Object {}}
              />
              <Text
                dataSet={Object {}}
                style={
                  Object {
                    "backgroundColor": undefined,
                    "color": "#171717",
                    "fontFamily": undefined,
                    "fontSize": 18,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                    "lineHeight": 27,
                    "marginBottom": 8,
                    "marginTop": 8,
                    "textDecorationLine": undefined,
                  }
                }
              >
                undefined undefined
              </Text>
            </View>
          </View>
        </View>
        <RCTScrollView
          contentContainerStyle={
            Array [
              Object {},
              Object {
                "dataSet": Object {},
              },
            ]
          }
          data={
            Array [
              Object {
                "icon": <Icon
                  allowFontScaling={false}
                  color="#000000"
                  name="power-off"
                  size={24}
                />,
                "id": "LOGOUT",
                "label": "LOGOUT",
                "onPress": [Function],
              },
            ]
          }
          dataSet={Object {}}
          getItem={[Function]}
          getItemCount={[Function]}
          keyExtractor={[Function]}
          onContentSizeChange={[Function]}
          onLayout={[Function]}
          onMomentumScrollBegin={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          removeClippedSubviews={false}
          renderItem={[Function]}
          scrollEventThrottle={50}
          stickyHeaderIndices={Array []}
          style={Object {}}
          viewabilityConfigCallbackPairs={Array []}
        >
          <View>
            <View
              onLayout={[Function]}
              style={null}
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  dataSet={Object {}}
                  style={
                    Array [
                      Object {
                        "borderBottomWidth": 1,
                        "borderColor": "#7f7f7f",
                        "paddingBottom": 8,
                        "paddingLeft": 16,
                        "paddingRight": 20,
                        "paddingTop": 8,
                      },
                      Array [
                        Object {},
                      ],
                    ]
                  }
                >
                  <View
                    dataSet={Object {}}
                    style={
                      Object {
                        "flexDirection": "row",
                      }
                    }
                  >
                    <Text
                      allowFontScaling={false}
                      selectable={false}
                      style={
                        Array [
                          Object {
                            "color": "#000000",
                            "fontSize": 24,
                          },
                          undefined,
                          Object {
                            "fontFamily": "FontAwesome",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          Object {},
                        ]
                      }
                    >
                      ï€‘
                    </Text>
                    <View
                      dataSet={Object {}}
                      style={
                        Object {
                          "width": 12,
                        }
                      }
                    />
                    <View
                      dataSet={Object {}}
                      style={
                        Object {
                          "justifyContent": "center",
                        }
                      }
                    >
                      <Text
                        dataSet={Object {}}
                        style={
                          Object {
                            "backgroundColor": undefined,
                            "color": "#171717",
                            "fontFamily": undefined,
                            "fontSize": 14,
                            "fontStyle": "normal",
                            "fontWeight": "700",
                            "letterSpacing": 0,
                            "lineHeight": 21,
                            "textDecorationLine": undefined,
                          }
                        }
                      >
                        LOGOUT
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RCTScrollView>
      </View>
      <View
        pointerEvents="none"
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
</RNCSafeAreaProvider>
`;
